### PR TITLE
feat: ensure signatory emails are normalized

### PR DESCRIPTION
**Pull request by guyjackson-ch**
## Description

This pull request improves data consistency by normalizing email addresses for directors—ensuring they are always trimmed and lowercased—when mapping or updating director data. It also introduces new test data builders and parameterized tests to verify this normalization behavior.

**Data normalization improvements:**
* Updated `DissolutionRequestMapper` and `DissolutionDirectorPatcher` to trim and lowercase email addresses before storing them, ensuring consistent email formatting. [[1]](diffhunk://#diff-4a5b911437bc3bfb7763dbf05c152c8c39864c611d85c6ad6d53c0f0d4661ea7L70-R69) [[2]](diffhunk://#diff-6736d7555eb3f8fd7df7b87126bd3a30aa1bf328353093482d1cd2fc6c1783beL57-R57)

**Test enhancements:**
* Added parameterized tests in `DissolutionDirectorPatcherTest` to verify email normalization (trimming and lowercasing).
* Introduced `DissolutionDirectorPatchRequestTestDataBuilder` and `DissolutionDirectorTestDataBuilder` utility classes for easier test data creation. [[1]](diffhunk://#diff-a2720288b4dbf8fc743a6d695b1f0733071bf06adb0c3a01f504f7c653ed7cbbR1-R31) [[2]](diffhunk://#diff-fabe50631f78a7cf7fe30cd045611812caeb19146a8cffc42a4148d5bfa4deb1R1-R45)
* Refactored tests to use new test data builders for better readability and maintainability. [[1]](diffhunk://#diff-209097632b75f8d4c1bc17362b9b967ba961447959ab62ee88c83c9df6af02dfL157-R168) [[2]](diffhunk://#diff-209097632b75f8d4c1bc17362b9b967ba961447959ab62ee88c83c9df6af02dfL23-R37)

## Jira Ticket

[fs-372](https://companieshouse.atlassian.net/browse/FS-372)

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorPatchRequestTestDataBuilder.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        this.email = email;
+        return this;
+    }
+
+    public DissolutionDirectorPatchRequestTestDataBuilder withOnBehalfName(String onBehalfName) {
+        this.onBehalfName = onBehalfName;
+        return this;
+    }
+
+    public DissolutionDirectorPatchRequest build() {
```
Comment (gsraich) [inline src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorPatchRequestTestDataBuilder.java:24]: It seems a similar test fixture exists (`src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorFixtures.java`), but it doesn't leverage the builder pattern like this test data builder, perhaps this should be consolidated, in a subsequent PR if it's quick

File: src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorTestDataBuilder.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        this.email = email;
+        return this;
+    }
+
+    public DissolutionDirectorTestDataBuilder withOnBehalfName(String onBehalfName) {
+        this.onBehalfName = onBehalfName;
+        return this;
+    }
+
+    public DissolutionDirector build() {
```
Comment (gsraich) [inline src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorTestDataBuilder.java:43]: Similar to my previous comment, `generateDissolutionDirector` exists as a fixture, but it isn't flexible like this builder, something to consolidate later and not in this PR

File: src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java [hunk 3]
```diff
@@ -124,12 +124,12 @@ public void mapToDissolution_setsDirectorsToSignFromRequestBody() {
 
         final DirectorRequest selectedDirector1 = DissolutionFixtures.generateDirectorRequest();
         selectedDirector1.setOfficerId(officerId1);
-        selectedDirector1.setEmail("DIRECTOR@mail.com");
+        selectedDirector1.setEmail(" DIRECTOR@mail.com");
```
Comment (gsraich) [inline src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java:127]: I'm just thinking about in the future, if someone is trying to understand the rules of this system, is the reasoning behind lowercasing and trimming the email clear? I think it might be a good idea to have a comment or have a test named something like "ensures email addresses are handled consistently regardless of letter casing".
Because this whole piece of work exists because this validation step was missed.

Also the test will pass if they accidentally or intentionally remove the whitespace or change the casing.

**Review by gsraich — CHANGES_REQUESTED** [review]
Looks good, just a small thought I had, happy to ignore and approve if you think its OTT

File: src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java (hunk 3 above)
Comment (gsraich) [reply src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java:127]: I probably should say that before this change it wasn't clear why the emails were lowercased, and because they are saved as-is, and there wasn't defensive code around the emails, this became a bug in the system.

At least that is my understanding of FS-372, i could be misinterpreting the ticket

File: src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorPatchRequestTestDataBuilder.java (hunk 1 above)
Comment (guyjackson-ch) [reply src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorPatchRequestTestDataBuilder.java:24]: I've removed the DissolutionDirectorFixtures

I originally retained it because I wanted to limit the extent of the changes.

File: src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorTestDataBuilder.java (hunk 2 above)
Comment (guyjackson-ch) [reply src/test/java/uk/gov/companieshouse/fixtures/DissolutionDirectorTestDataBuilder.java:43]: I have proxied the call to the builder. To migrate fully would create a lot of changes so will leave this for now.

Future changes can migrate incrementally as required.

File: src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java (hunk 3 above)
Comment (guyjackson-ch) [reply src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java:127]: So I think I originally took the view that it was more effort than it was worth for now. I agree with the intent of your comment however.

But to do this elegantly without causing a bloated additional test is a bit more than I want to do for this ticket at this stage.

Comment (gsraich) [reply src/test/java/uk/gov/companieshouse/mapper/DissolutionRequestMapperTest.java:127]: Thats fair enough, this requirement / system rule is in the commit history once this PR gets merged, future devs can run git blame and see the intent

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by gsraich — APPROVED** [review]
LGTM 👍
